Adding Poster support for top 10 lists and user-specifc cards

mcclown opened the pull request:
One last big addition, for this year at least.

<img width="537" height="673" alt="image" src="https://github.com/user-attachments/assets/637742ed-e0b3-4d2f-b89b-4f30e3c336b5" />

Adding support for grabbing posters from Plex, caching them in Wrapperr and then using them as part of top 10 lists, as well as in user-specific cards.

The feature can be turned on/off and also has UI for clearing the cached poster from disk, when required. Plex posters are pre-fetched for all server-wide lists. Users opening their Wrapped triggers download of user-specific posters. Downloads of Plex posters happen in parallel, so they have minimal impact on load times.

mcclown committed:
Initial commit for poster support

mcclown committed:
Fixing up poster fetch logic

mcclown committed:
Fixing up poster styling for top 10 lists

mcclown committed:
Add posters support to special cards

mcclown committed:
Fixing bug that's caused by a typo with the parentRatingKey/grandParentRatingKey

mcclown committed:
Merge branch 'typoBugFix' into addPlexPosters

mcclown committed:
Noe tv shows use show posters instead of season posters

mcclown committed:
Removing debugging code

mcclown committed:
Cleaning up CSS

mcclown committed:
Adding support for clearing poster cache in UI

mcclown committed:
Fixing up config UI

mcclown committed:
Refactoring poster downloads

mcclown committed:
Small CSS tweak

aunefyren commented:
Not at all, sorry. I have been really swamped, but I expecting these can be instantly merged when I have time to look through them. Thank you so much for contributing.

aunefyren commented:
Wow, incredible works, this looks great. Never imaged the poster would add so much. Seems the highlighting/expansion of the items in the list is a bit broken. Only the top item expands, the rest don't. I would probably remove that feature or make it consistant on all items.

<img width="601" height="356" alt="image" src="https://github.com/user-attachments/assets/97f330b1-d09f-4505-a0a7-b88549b25cf7" />

aunefyren committed:
Merge branch 'main' into addPlexPosters

aunefyren committed:
Merge and CSS tweaks

aunefyren commented:
I did some CSS tweaking, so it is a bit better at least. Any clue why there is a bunch of 404's when getting posters? No issues on the front end side as I can see.

<img width="1557" height="482" alt="image" src="https://github.com/user-attachments/assets/649e8da5-65c2-4276-884c-da68c227906c" />

mcclown commented:
> I did some CSS tweaking, so it is a bit better at least. Any clue why there is a bunch of 404's when getting posters? No issues on the front end side as I can see.
> 
> <img alt="image" width="1557" height="482" src="https://private-user-images.githubusercontent.com/31650531/564813521-649e8da5-65c2-4276-884c-da68c227906c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzM3OTA1NjYsIm5iZiI6MTc3Mzc5MDI2NiwicGF0aCI6Ii8zMTY1MDUzMS81NjQ4MTM1MjEtNjQ5ZThkYTUtNjVjMi00Mjc2LTg4NGMtZGE2OGMyMjc5MDZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzE3VDIzMzEwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5YzgyNzc5MDNjNmQ2ZDYwMTY5Yzg5NzMzYTNhNTNjN2NkNDA5YWFmNGZiMTcxYzk1MzRlNDcyZmFmNGZkODImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5cs2SEx_B-PPaUYnIiRq3rBFovFcs6MUDg5dLwhr5Bg">

That's a good question. I'll take a look

aunefyren committed:
Merge remote-tracking branch 'origin/main' into pr/127

aunefyren committed:
fix: resolve music 404s, secure download route, add bounded queue

The PR had the frontend request poster art for music top-lists that the backend never downloads, producing a wall of 404s. This fixes that and addresses several issues found while reviewing the feature.
- Frontend: skip poster <img> tags for music lists in top_list() (the actual source of the reported 404s), add posterImgTag()/handlePosterError() helpers that download-then-retry on miss before falling back, and route all poster render sites through them.
- Security: require a valid session on POST /download/poster when Plex-Auth is enabled and validate thumb_path against the Tautulli metadata form (closes an unauthenticated SSRF / API-key abuse vector), harden the poster path-traversal check, write downloads to a temp file + atomic rename to avoid torn reads.
- Performance: replace the per-request preload and the unbounded per-request goroutine with a single bounded, app-lifetime worker pool with in-flight dedup, both.go now queue user + year posters non-blocking so the statistics response returns immediately.
- Cleanup: decouple poster preloading from UseCache (now gated solely on EnablePosters), memorize the Tautulli server hash, add Cache-Control on served posters, and remove the now-dead batch helpers.

aunefyren committed:
fix: add URL validation/regex

Hopefully CodeQL is happy
